pkg/controller/plan/migration.go

```diff
@@ -1439,10 +1456,29 @@ func (r *Migration) updateCopyProgress(vm *plan.VMStatus, step *plan.Step) (err
 				var importer *core.Pod
 				var found bool
 				var kErr error
-				if dv.PVC == nil {
+				if dv.PVC == nil && !r.Plan.IsSourceProviderOCP() {
 					found = false
 				} else {
-					importer, found, kErr = r.kubevirt.GetImporterPod(*dv.PVC)
+					if r.Plan.IsSourceProviderOCP() {
+						pvc := &core.PersistentVolumeClaim{}
+						err = r.Destination.Client.Get(context.TODO(), types.NamespacedName{
+							Namespace: r.Plan.Spec.TargetNamespace,
+							Name:      dv.Status.ClaimName,
+						}, pvc)
+						if err != nil {
+							log.Error(
+								err,
+								"Could not get PVC for DataVolume.",
+								"vm",
+								vm.String(),
+								"dv",
+								path.Join(dv.Namespace, dv.Name))
+							continue
+						}
+						importer, found, kErr = r.kubevirt.GetImporterPod(*pvc)
+					} else {
+						importer, found, kErr = r.kubevirt.GetImporterPod(*dv.PVC)
```

Comment:
I think it would be better to be more explicit here by modifying the changes above in a way that each block would set the pvc differently and then we proceed to a single call to GetImporterPod